javascript_nodejs/5.multiturn-prompts-bot/dialogs/mainDialog/index.js

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActionTypes, MessageFactory } = require('botbuilder');
+
+const { TextPrompt, NumberPrompt, DialogSet } = require('botbuilder-dialogs');
+
+const USER_NAME_PROP = 'user_name';
+const AGE_PROP = 'user_age';
+const WHO_ARE_YOU = 'who_are_you';
+const HELLO_USER = 'hello_user';
+
+class MainDialog {
+    /**
+     * 
+     * @param {Object} conversationState 
+     * @param {Object} userState 
+     */
+    constructor (conversationState, userState) {
+        var that = this;
+
+        // creates a new state accessor property. see https://aka.ms/about-bot-state-accessors to learn more about the bot state and state accessors 
+        this.conversationState = conversationState;
+        this.userState = userState;
+
+        this.userName = this.userState.createProperty(USER_NAME_PROP);
+        this.userAge = this.userState.createProperty(AGE_PROP);
+
+        this.dialogs = new DialogSet();
```

Comment:
This isn't good, this DialogSet doesn't have a `dialogState` StatePropertyAccessor so this shouldn't work...